src/map/script.c

```diff
@@ -1177,24 +1239,44 @@ static const char *parse_variable(const char *p)
 static bool is_number(const char *p)
 {
 	const char *np;
-	if (!p)
-		return false;
-	if (*p == '-' || *p == '+')
+	nullpo_retr(false, p);
+
+	if (*p == '-' || *p == '+') {
 		p++;
+	}
+
 	np = p;
+
 	if (*p == '0' && p[1] == 'x') {
-		p+=2;
-		np = p;
-		// Hexadecimal
-		while (ISXDIGIT(*np))
+		// Hexadecimal: 0xFFFF
+		np = p += 2;
```

Comment:
```suggestion
		np = (p += 2);
```
Or split it into separate operations.